### DISCUSSION TITLE
context.report doesn't work within a .then statement

### DISCUSSION THREAD

**Discussion by Brian-Bartels** [eslint/eslint#14998] [Q&A] [answered]
I am attempting to make an ESLint rule that aggregates the results of other ESLint rules. This is to create a mimic of the maintainability index. I am doing this by creating an ESLint object within the create method of my new rule. I then call the ESLint object on the source code of the context. However, in the .then of the results I try to report violations but none of them seem to be going through. My rule is definitely bending things to the extreme but I suspect that I'm not the first to think of aggregate rules

For example:
message: "this file has too many complexity violations"
when there are many instances of max-depth and max-statements and max-params

**Comment by nzakas** [comment, marked as answer]
Rules are assumed to be sync, so anything you do in a then() will actually occur after ESLint has already processed everything. There isn’t a way to do what you’re attempting at the moment.
